### ISSUE TITLE
[BLOCK IO] Lift the `H != 1` guard on 1D strided stores

### ISSUE THREAD

**Issue by NathanVoldman** [intel/intel-xpu-backend-for-triton#7805]
## Summary

`reshape1DStridedStore` bails out whenever the detected tile has more than one row:

```cpp
if (info->H != 1) {
  LDBG("H=" << H << " > 1 not yet supported, skip 1D reshape");
  return;
}
```

This was landed as an explicit workaround in #6682 to stop an accuracy regression (#6634), and approved as provisional fix.
This task is to work on improving the pass to see if we can allow the case H > 1

The guard is a proxy, not a precondition.
A 2D block store needs the value in hardware delivery order - lane *k* owns column *k*, registers stack rows. `tt.reshape`'s *inferred* 2D encoding is the opposite. At `H == 1` the two encodings coincide - and that's why this change mostly hide the bug.

**Comment by NathanVoldman** [comment]
Gain in benchmark from [6532](https://github.com/intel/intel-xpu-backend-for-triton/issues/6532
)
Tested on BMG (B65)

| config | baseline | With change | Δ time | path change |
|---|---|---|---|---|
| W=32 S=96 f16 ~100M | 1.7294 ms | 1.6740 ms| −3.2% | fallback → 2Dblockstore |
| W=32 S=192 f16 ~100M | 2.0545 ms | 1.9492 ms | −5.1% | fallback → 2Dblockstore |
| W=32 S=96 f16 ~10M | 0.2028 ms | 0.1904 ms | −6% | fallback → 2Dblockstore |
| W=32 S=192 f16 ~10M | 0.2259 ms | 0.2181 ms | −3.5% | fallback → 2Dblockstore |
| W=32 S=128 f16 ~10M | 0.2206 ms | 0.2223 ms | +0.8% | fallback → 2Dblockstore |
| W=16 S=48 f16 10M | 0.2244 ms | 0.2216 ms | −1.2% | *control* (fallback)|
| W=16 S=96 f16 10M | 0.3389 ms | 0.3360 ms | −0.9% | *control* (fallback)|
| W=32 S=96 f32 ~10M | 0.3371 ms | 0.3377 ms | +0.2% | *control* (fallback)|
| W=32 S=192 f32 ~10M | 0.3464 ms | 0.3476 ms | +0.3% | *control* (fallback)|


Benchmark:
```python
"""Benchmark 1D strided stores at scale — no mask (splat true).

Matches the Inductor pattern where xnumel is a multiple of XBLOCK,
so the mask is always true and the reshape can trigger.
"""
import triton
import triton.language as tl
import torch
import numpy as np


@triton.jit
def strided_store_nomask(
    in_ptr, out_ptr,
    xnumel,
    W: tl.constexpr, S: tl.constexpr,
    XBLOCK: tl.constexpr,
):
    xoffset = tl.program_id(0) * XBLOCK
    xindex = xoffset + tl.arange(0, XBLOCK)
    xmask = tl.full([XBLOCK], True, tl.int1)
    val = tl.load(in_ptr + xindex, xmask).to(tl.float32)
    result = val * 0.5  # simple compute to avoid being optimized away
    result = result.to(tl.float16)
    out_addr = (xindex % W) + (xindex // W) * S
    tl.store(out_ptr + out_addr, result, xmask)


def bench_config(W, S, dtype, num_rows, xblock, num_warps, label):
    xnumel = W * num_rows
    assert xnumel % xblock == 0, f"xnumel={xnumel} not divisible by XBLOCK={xblock}"
    in_tensor = torch.randn(xnumel, device='xpu', dtype=dtype)
    out_tensor = torch.zeros(num_rows * S, device='xpu', dtype=dtype)

    grid = (xnumel // xblock,)

    # Warmup
    for _ in range(10):
        strided_store_nomask[grid](
            in_tensor, out_tensor, xnumel,
            W=W, S=S, XBLOCK=xblock, num_warps=num_warps,
        )
    torch.xpu.synchronize()

    # Benchmark
    times = []
    for _ in range(100):
        start = torch.xpu.Event(enable_timing=True)
        end = torch.xpu.Event(enable_timing=True)
        start.record()
        strided_store_nomask[grid](
            in_tensor, out_tensor, xnumel,
            W=W, S=S, XBLOCK=xblock, num_warps=num_warps,
        )
        end.record()
        torch.xpu.synchronize()
        times.append(start.elapsed_time(end))

    ms = np.median(times)
    elem_size = in_tensor.element_size()
    num_bytes = xnumel * elem_size * 2
    gb_per_s = (num_bytes / 1e9) / (ms / 1e3)
    print(f"  {label:55s}  {ms:.4f}ms  {gb_per_s:8.2f} GB/s")
    return ms, gb_per_s


def main():
    print("=" * 90)
    print("Benchmark: 1D strided stores — no mask (Inductor pattern)")
    print("=" * 90)

    configs = [
        # (W, S, dtype, num_rows, xblock, num_warps, label)
        # xnumel must be divisible by xblock

        # W=32 variants (original Inductor case)
        (32, 96,  torch.float16, 307200, 1024, 4, "W=32 S=96  f16 ~10M (original pattern)"),
        (32, 128, torch.float16, 307200, 1024, 4, "W=32 S=128 f16 ~10M"),
        (32, 192, torch.float16, 307200, 1024, 4, "W=32 S=192 f16 ~10M"),

        # Narrow width — partial cache lines
        (16, 48,  torch.float16, 640000, 1024, 4, "W=16 S=48  f16 10M (half cache line)"),
        (16, 96,  torch.float16, 640000, 1024, 4, "W=16 S=96  f16 10M"),

        # f32 (larger element)
        (32, 96,  torch.float32, 307200, 1024, 4, "W=32 S=96  f32 ~10M"),
        (32, 192, torch.float32, 307200, 1024, 4, "W=32 S=192 f32 ~10M"),

        # Very large
        (32, 96,  torch.float16, 3072000, 1024, 4, "W=32 S=96  f16 ~100M"),
        (32, 192, torch.float16, 3072000, 1024, 4, "W=32 S=192 f16 ~100M"),
    ]

    for W, S, dtype, num_rows, xblock, num_warps, label in configs:
        try:
            bench_config(W, S, dtype, num_rows, xblock, num_warps, label)
        except Exception as e:
            print(f"  {label:55s}  FAILED: {e}")


if __name__ == '__main__':
    main()
```

**Comment by NathanVoldman** [comment]
Gain even improved following some changes raised in the CR:

| Config | Baseline (ms) | Optimized (ms) | Δ ms | gain % |
|---|---|---|---|---|
| W=32 S=96 ~10M | 0.2199 | 0.1870 | **−0.0329** | **+18%** |
| W=32 S=128 ~10M | 0.2204 | 0.2053 | −0.0151 | +7% |
| W=32 S=192 ~10M | 0.2274 | 0.2140 | −0.0134 | +6% |
| W=32 S=96 ~100M | 1.7263 | 1.6627 | **−0.0636** | +4% |
| W=32 S=192 ~100M | 2.0526 | 1.9415 | −0.1111 | +6% |

**Comment by NathanVoldman** [comment]
## Latest Benchmark Results

**Method:** 10 independent runs per condition, cache (`~/.triton/cache`) cleared before each run.
Trimmed mean: min and max dropped per config, mean of remaining 8 runs reported.  

| Config | Main ms | Branch ms | Δ ms | Δ gain% | Tag |
|---|---|---|---|---|---|
| W=32 S=96  f16 ~10M | 0.1987 | 0.1860 | −0.0127 | **+6.4%** | ✅ changed |
| W=32 S=128 f16 ~10M | 0.2173 | 0.2055 | −0.0118 | **+5.5%** | ✅ changed |
| W=32 S=192 f16 ~10M | 0.2226 | 0.2131 | −0.0095 | **+4.3%** | ✅ changed |
| W=32 S=96  f16 ~100M | 1.7224 | 1.6620 | −0.0604 | **+3.5%** | ✅ changed |
| W=32 S=192 f16 ~100M | 2.0491 | 1.9362 | −0.1128 | **+5.5%** | ✅ changed |
| W=16 S=48  f16 10M | 0.2137 | 0.2125 | −0.0012 | +0.6% | ⚙ control (W<tpw) |
| W=16 S=96  f16 10M | 0.3320 | 0.3289 | −0.0031 | +0.9% | ⚙ control (W<tpw) |
| W=32 S=96  f32 ~10M | 0.3317 | 0.3311 | −0.0006 | +0.2% | ⚙ control (f32: W>elem limit) |
| W=32 S=192 f32 ~10M | 0.3424 | 0.3414 | −0.0010 | +0.3% | ⚙ control (f32: W>elem limit) |
